features/000-anonymous-user/design.md

```diff
@@ -0,0 +1,112 @@
+# Anonymous User
+
+## Objective
+- Allow users to signup without using a real identity (e.g. login ID, OAuth).
+- Allow anonymous users to authenticate securely.
+- Allow promoting anonymous users to normal users.
+
+## Design
+
+### Anonymous identity
+
+A new identity type `anonymous` would be added. It is expected that users would
+either have no anonymous identity, or have exactly one anonymous identity
+without other types of identity.
+
+Anonymous identity has fields:
+- public key: the public key represented as a JWK.
+- key ID: a random string (`^[-\w]{8,64}$`), used for efficient lookup.
+
+### Authentication
+
+To signup as anonymous user:
```

Comment:
Is this flow compatible with WebAuthn register flow?

Reply:
Not necessarily. WebAuthn has different security properties as anonymous user, most notably the key pair may not be stored securely on web platform. Also, WebAuthn can be used on web platform only.

Reply:
My main concern is that we do not invent a protocol ourselves but reference to a similar and existing protocol which has been studied, especially in the security aspect.

Reply:
The anonymous user flows are largely taken from WebAuthn. However it is not fully compatible with WebAuthn for simplicity, e.g. using JSON instead of CBOR, not distinguishing between attestation and assertion, etc.